cpp/src/gandiva/precompiled/extended_math_ops_test.cc

```diff
@@ -120,6 +120,50 @@ TEST(TestExtendedMathOps, TestRoundDecimal) {
   VerifyFuzzyEquals(round_float64_int32((double)INT_MIN - 1, 0), (double)INT_MIN - 1);
 }
 
+TEST(TestExtendedMathOps, TestBRoundDecimal) {
+  EXPECT_FLOAT_EQ(bround_float32(0.0f), 0);
+  EXPECT_FLOAT_EQ(bround_float32(2.5f), 2);
+  EXPECT_FLOAT_EQ(bround_float32(3.5f), 4);
+  EXPECT_FLOAT_EQ(bround_float32(-2.5f), -2);
+  EXPECT_FLOAT_EQ(bround_float32(-3.5f), -4);
+  EXPECT_FLOAT_EQ(bround_float32(1.4999999f), 1);
+  EXPECT_EQ(std::signbit(bround_float32(0)), 0);
+  EXPECT_EQ(std::signbit(bround_float64(0)), 0);
+  EXPECT_FLOAT_EQ(bround_float32_int32(8.25f, 1), 8.2f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(8.35f, 1), 8.4f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(6.225f, 2), 6.22f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(8.335f, 2), 8.34f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(-8.25f, 1), -8.2f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(-8.35f, 1), -8.4f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(-6.225f, 2), -6.22f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(-8.335f, 2), -8.34f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(-6.225f,-2), -6.22f);
+  EXPECT_FLOAT_EQ(bround_float32_int32(8.335f, -2), 8.34f);
+
+  VerifyFuzzyEquals(bround_float32(2.5f), 2);
+  VerifyFuzzyEquals(bround_float32(3.5f), 4);
+  VerifyFuzzyEquals(bround_float32(-2.5f), -2);
+  VerifyFuzzyEquals(bround_float32(-3.5f), -4);
+  VerifyFuzzyEquals(bround_float32(1.4999999f), 1);
+  VerifyFuzzyEquals(bround_float64(2.5), 2);
+  VerifyFuzzyEquals(bround_float64(3.5), 4);
+  VerifyFuzzyEquals(bround_float64(-2.5), -2);
+  VerifyFuzzyEquals(bround_float64(-3.5), -4);
+  VerifyFuzzyEquals(bround_float64(1.4999999), 1);
```

Comment:
Please add a test for bround_float64(1.50001) = 2

Reply:
Test added